File: backend-java/src/test/java/com/financepulse/engine/testsupport/SequentialIdGenerator.java
package com.financepulse.engine.testsupport;

import com.financepulse.engine.application.ports.IdGenerator;

public class SequentialIdGenerator implements IdGenerator {

    private final String prefix;
    private int counter = 0;

    public SequentialIdGenerator() {
        this("id");
    }

    public SequentialIdGenerator(String prefix) {
        this.prefix = prefix;
    }

    @Override
    public String generate() {
        counter += 1;
        return prefix + "-" + counter;
    }
}

File: backend/src/test-support/SequentialIdGenerator.ts
import type { IdGenerator } from "../application/ports/IdGenerator";

export class SequentialIdGenerator implements IdGenerator {
  private counter = 0;

  constructor(private readonly prefix: string = "id") {}

  generate(): string {
    this.counter += 1;
    return `${this.prefix}-${this.counter}`;
  }
}
